span.go

```diff
@@ -119,9 +119,13 @@ func (s *spanImpl) FinishWithOptions(opts opentracing.FinishOptions) {
 	if s.tracer.Options.DebugAssertUseAfterFinish {
 		// This makes it much more likely to catch a panic on any subsequent
 		// operation since s.tracer is accessed on every call to `Lock`.
+		pool := s.tracer.spanPool
```

Comment:
It's not necessary here because the tracer never changes. But I prefer if an object coming out of a buffer pool is the same as an object newly allocated, i.e. completely zero.